flang/lib/Lower/ConvertCall.cpp

```diff
@@ -1158,6 +1181,7 @@ genUserCall(Fortran::lower::PreparedActualArguments &loweredActuals,
     case PassBy::CharBoxValueAttribute:
     case PassBy::Box:
     case PassBy::BaseAddress:
+    case PassBy::BoxProcRef:
```

Comment:
After I fixed an `assert` to allow `BoxProcRef`, the above test case is lowered to 
```
  func.func @_QQmain() {
    %0 = fir.zero_bits !fir.ref<none>
    %1 = fir.convert %0 : (!fir.ref<none>) -> !fir.ref<!fir.boxproc<() -> ()>>
    fir.call @_QPtest(%1) fastmath<contract> : (!fir.ref<!fir.boxproc<() -> ()>>) -> ()
    return
  }

```

Would this be good?

Reply:
No it would lead to the same LLVM as `fir.absent !fir.ref<!fir.boxproc<() -> ()>>`. I think you need to somehow create a `fir.emboxproc %null` (createNullBoxProc) in this case instead.

Reply:
Ok. Got it. I will create a temp BoxProcType that is zeroed.

Reply:
With the latest update, it generates:
```
    %0 = fir.alloca !fir.boxproc<() -> ()>
    %1 = fir.zero_bits () -> ()
    %2 = fir.emboxproc %1 : (() -> ()) -> !fir.boxproc<() -> ()>
    fir.store %2 to %0 : !fir.ref<!fir.boxproc<() -> ()>>
    fir.call @_QPtest(%0) fastmath<contract> : (!fir.ref<!fir.boxproc<() -> ()>>) -> ()
```